### PR TITLE
fix: upgrade apecloud mysql addon

### LINKED ISSUE #543: [Improvement] upgrade apecloud mysql to 0.9

### PR DESCRIPTION
**Pull request by shanshanying**
- fix #543
- List of cmpds:
image
- cluster def with topologies::c:q
image

create a cluster with proxy:
image

### CODE REVIEW

File: addons/apecloud-mysql/templates/cmpdversion.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+kind: ComponentVersion
+metadata:
+  name: apecloud-mysql
+  labels:
+    {{- include "apecloud-mysql.labels" . | nindent 4 }}
+spec:
+  releases:
+  - name: apecloud-mysql-8.0.30
+    serviceVersion: "8.0.30"
+    images:
```
Comment (Y-Rookie) [inline addons/apecloud-mysql/templates/cmpdversion.yaml:11]: In the release, it is best to specify the images as static.

File: addons/apecloud-mysql/templates/cmpd-apecloudmysql.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+apiVersion: apps.kubeblocks.io/v1alpha1
+kind: ComponentDefinition
+metadata:
+  name: {{ include "apecloud-mysql.componentDefName" . }}
```
Comment (Y-Rookie) [inline addons/apecloud-mysql/templates/cmpd-apecloudmysql.yaml:4]: Does apecloud-mysql need to support multiple versions? If multi-version support is required, it is recommended to reconsider the naming of cmpd.

File: addons/apecloud-mysql/templates/cmpdversion.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  compatibilityRules:
+  - releases: [apecloud-mysql-8.0.30]
+    compDefs: [apecloud-mysql]
+  - releases: [apecloud-mysql-auditlog-8.0.30]
+    compDefs: [apecloud-mysql-auditlog]
+---
+apiVersion: apps.kubeblocks.io/v1alpha1
+kind: ComponentVersion
+metadata:
+  name: wescale-vtcontroller
```
Comment (leon-inf) [inline addons/apecloud-mysql/templates/cmpdversion.yaml:31]: It seems that it's good enough to provide a component version for wescale.

File: addons/apecloud-mysql/templates/_helpers.tpl [hunk 4]
```diff
@@ -77,4 +77,278 @@ Backup Tool image
...
+
+{{- define "apecloud-mysql.spec.runtime.volumes" -}}
+{{- if .Values.logCollector.enabled }}
+- name: log-data
+  hostPath:
+    path: /var/log/kubeblocks
+    type: DirectoryOrCreate
+{{- end }}
+- name: annotations
+  downwardAPI:
```
[the remark this answers is not in the document]
Comment (shanshanying) [reply addons/apecloud-mysql/templates/_helpers.tpl:346]: such change will be submitted in a follow-up PR (remove dependent on KB_REPLICA_COUNT, KB_i_HOSTNAME, etc)

File: addons/apecloud-mysql/templates/_helpers.tpl [hunk 5]
```diff
@@ -77,4 +77,278 @@ Backup Tool image
...
+      image: {{ .Values.image.registry | default "docker.io" }}/{{ .Values.image.repository }}:{{ default .Values.image.tag }}
+      exec:
+        command:
+          - /scripts/switchover-with-candidate.sh
+    withoutCandidate:
+      image: {{ .Values.image.registry | default "docker.io" }}/{{ .Values.image.repository }}:{{ default .Values.image.tag }}
+      exec:
+        command:
+          - /scripts/switchover-without-candidate.sh
+    scriptSpecSelectors:
```
[the remark this answers is not in the document]
Comment (shanshanying) [reply addons/apecloud-mysql/templates/_helpers.tpl:183]: removed

File: addons/apecloud-mysql/templates/cmpdversion.yaml (hunk 3 above)
Comment (shanshanying) [reply addons/apecloud-mysql/templates/cmpdversion.yaml]: there are two components for wescale,  vectonroller and vtgate, so use two cmpv, one for each.

File: addons/apecloud-mysql/templates/cmpd-apecloudmysql.yaml (hunk 2 above)
Comment (shanshanying) [reply addons/apecloud-mysql/templates/cmpd-apecloudmysql.yaml:4]: it only supports mysql 8 (major version). so we keep it as it is.

File: addons/apecloud-mysql/templates/cmpdversion.yaml (hunk 3 above)
Comment (shanshanying) [reply addons/apecloud-mysql/templates/cmpdversion.yaml]: fixed

**Review by Y-Rookie — APPROVED** [review]

File: addons/apecloud-mysql/templates/_helpers.tpl (hunk 4 above)
[the remark this answers is not in the document]
Comment (shanshanying) [reply addons/apecloud-mysql/templates/_helpers.tpl:344]: issues: https://github.com/apecloud/kubeblocks-addons/issues/551

**Review by leon-inf — APPROVED** [review]

**Pull request merged**